src/il_representations/algos/encoders.py

```diff
@@ -156,6 +157,39 @@ def do_warning(message):
     ]
 }
 
+class CarlaMLP(nn.Module):
+    """Use MLP for encodder network, with structure defined by model_arch.
+       Its default follows https://github.com/rail-berkeley/d4rl_evaluations/b
+       lob/c93b9d2bf443cd5b8c43542fca9937dc50e630c3/brac/behavior_regularized_
+       offline_rl/brac/networks.py#L39.
+    """
+    def __init__(self, observation_space, representation_dim, model_arch,
+                 flatten_input=False):
+        super().__init__()
+        if flatten_input:
+            # Assuming the input has frame stack. We only use the last RGB frame.
```

Comment:
This is a bit dangerous—it's silently throwing away frame stacking, which might cause this architecture to mysteriously fail in partially observable domains.
Why not flatten directly, without this behaviour? It won't affect CARLA at all (since we're not stacking frames), and is less likely to cause confusing results in other domains.